shared/output/gapic/templates/garbage/test/so/much/trash/garbage_service_test.rb

```diff
@@ -386,7 +386,7 @@ def test_get_typical_garbage
       assert_equal Gapic::Protobuf.coerce({}, to: Google::Protobuf::Duration), request.duration
       assert_equal Gapic::Protobuf.coerce({}, to: So::Much::Trash::GarbageMap), request.msg
       assert_equal :Default, request.enum
-      assert_kind_of So::Much::Trash::TypicalGarbage::AmapEntry, request.amap.first
+      assert_equal Gapic::Protobuf.coerce({}, to: So::Much::Trash::TypicalGarbage::AmapEntry), request.amap
```

Comment:
This doesn't seem right either. `request.amap` should be some kind of a Protobuf Map type, not a single AmapEntry. I think we need a special case for `field.map?`.